internal/trafficpolicy/policy.go

```diff
@@ -78,11 +81,48 @@ func NewTrafficPolicy() *TrafficPolicy {
 	}
 }
 
+// validTrafficPolicyKeys are the only keys allowed at the top level of a traffic policy document.
+// Any other keys indicate a malformed policy that could lead to security issues if silently ignored.
+var validTrafficPolicyKeys = map[string]bool{
+	"on_http_request":  true,
+	"on_http_response": true,
+	"on_tcp_connect":   true,
+}
+
 // NewTrafficPolicyFromJSON creates a new TrafficPolicy from a JSON byte array.
+// It validates that all top-level keys are known phase names and returns an error
+// if unknown keys are present (which would otherwise be silently ignored).
 func NewTrafficPolicyFromJSON(data []byte) (*TrafficPolicy, error) {
+	if len(data) == 0 {
+		return NewTrafficPolicy(), nil
+	}
+
+	// First unmarshal to a map to detect unknown keys
+	var rawPolicy map[string]json.RawMessage
+	if err := json.Unmarshal(data, &rawPolicy); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal traffic policy: %w. raw traffic policy: %v", err, rawPolicy)
```

Comment:
```suggestion
		return nil, fmt.Errorf("failed to unmarshal traffic policy: %w. raw traffic policy: %v", err, data)
```

If it fails to unmarshal, it looks like we'll just print the empty rawPolicy as opposed to what we are trying to unmarshal.